python/ray/tune/experiment.py

```diff
@@ -121,7 +121,20 @@ def __init__(
                     "checkpointable function. You can specify checkpoints "
                     "within your trainable function."
                 )
-        self._run_identifier = Experiment.register_if_needed(run)
+        try:
+            self._run_identifier = Experiment.register_if_needed(run)
+        except RpcError as e:
+            if e.code() == grpc.StatusCode.RESOURCE_EXHAUSTED:
+                raise TuneError(
+                    "The Trainable/training function is too large for grpc resource "
+                    "limit. Check that its definition is not implicitly capturing a "
+                    "large array or other object in scope. "
+                    "Tip: use tune.with_parameters() to put large objects "
+                    "in the Ray object store."
+                )
```

Comment:
We should include the original exception in the message as well imo. Something like `\nOriginal exception: {traceback.format_exc()}`